src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_ELEMENT_MAX_SIZE = 100;
+    private static final int FIGURE_NUMBER = 5;
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private static final int DEFAULT_RADIUS = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(FIGURE_NUMBER)) {
+            case 0:
+                return getRandomSquare();
+            case 1:
+                return getRandomRectangle();
+            case 2:
+                return getRandomRightTriangle();
+            case 3:
+                return getRandomCircle();
+            default:
+                return getRandomIsoscelesTrapezoid();
+        }
+    }
+
+    private Figure getRandomSquare() {
+        String color = colorSupplier.getRandomColor();
+        int side = random.nextInt(FIGURE_ELEMENT_MAX_SIZE) + 1;
```

Comment:
`random.nextInt(FIGURE_ELEMENT_MAX_SIZE) + 1;` - it is logic (because of **+1**) and it repeats many times.
Create private method `getRandomLength()` for this.